game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/firststrike/DefensiveFirstStrike.java

```diff
@@ -0,0 +1,80 @@
+package games.strategy.triplea.delegate.battle.steps.fire.firststrike;
+
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.FIRST_STRIKE_UNITS_FIRE;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SELECT_FIRST_STRIKE_CASUALTIES;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.MustFightBattle.ReturnFire;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DefensiveFirstStrike implements BattleStep {
+
+  private static final long serialVersionUID = 3646211932844911163L;
+
+  protected final BattleState battleState;
+
+  protected final BattleActions battleActions;
+
+  protected final FirstStrikeStepOrder order;
+
+  protected final ReturnFire returnFire;
+
+  public DefensiveFirstStrike(
```

Comment:
Have you considered using Lombok @AllArgsConstructor to replace this constructor? The annotation should do the same thing.